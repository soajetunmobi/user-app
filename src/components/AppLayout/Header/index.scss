.header {
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding-left: 20px;
    padding-right: 50px;
}

.logo {
    font-size: 2rem;
    text-align: center;
    line-height: var(--logo-height);
    height: var(--logo-height);
    margin: 0;

    & > a{
        text-decoration: none;
    }
}
