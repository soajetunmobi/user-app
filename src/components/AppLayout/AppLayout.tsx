import React, { FunctionComponent, PropsWithChildren, ReactNode } from 'react'
import { Header } from './Header'
import { Main } from './Main'
import { Footer } from './Footer'

export const AppLayout: FunctionComponent = ({ children }: PropsWithChildren<ReactNode>) => {
  return (
    <>
      <Header />
      <Main>{children}</Main>
      <Footer />
    </>
  )
}
