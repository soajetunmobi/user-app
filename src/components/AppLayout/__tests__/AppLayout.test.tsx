import React from 'react'
import renderer from 'react-test-renderer'
import { AppLayout } from '../AppLayout'

test('it renders correctly', () => {
  const root = renderer.create(<AppLayout />).toJSON()
  expect(root).toMatchSnapshot()
})
