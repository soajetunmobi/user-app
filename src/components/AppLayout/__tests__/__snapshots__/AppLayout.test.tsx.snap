// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
Array [
  <div
    className="header"
  >
    <div
      className="logo"
    >
      <span
        className="cursor--pointer"
        data-testid="logo-link"
        onClick={[Function]}
      >
        User App
      </span>
    </div>
    <div>
      <span
        className="cursor--pointer"
        data-testid="login-link"
        onClick={[Function]}
      >
        Login |
         
      </span>
      <span
        className="cursor--pointer"
        onClick={[Function]}
      >
        Profile
      </span>
    </div>
  </div>,
  <main
    className="content"
  />,
  <footer
    className="footer"
  >
    <span>
      User App
    </span>
    <br />
    2020
  </footer>,
]
`;
